go/vt/tabletserver/query_splitter.go

```diff
@@ -176,9 +199,9 @@ func (qs *QuerySplitter) getWhereClause(start, end sqltypes.Value) *sqlparser.Wh
 			}
 		}
 	}
-	if qs.sel.Where != nil {
+	if whereClause != nil {
 		clauses = &sqlparser.AndExpr{
-			Left:  qs.sel.Where.Expr,
+			Left:  whereClause.Expr,
```

Comment:
Almost missed this. You need to parenthesize this using https://github.com/youtube/vitess/blob/master/go/vt/sqlparser/ast.go#L515.